File: ООП/Homework2/all_animals/hen.java
package all_animals;

public class hen extends birds{

    protected hen(int height, int weight, String eyes_color, int flight_altitude) {
        super(height, weight, eyes_color, flight_altitude);
    }

    public void fly() {
        String.format("Высота полёта: %d метров", super.flight_altitude);
    }

    public void voice() {
        System.out.println("\nКу-ка-ре-ку");
    }

    public String toString(){
        return String.format("Курица: %s", super.toString());
    }
   
}

File: ООП/Homework2/animals_PY/classes/birds/hen.py
from ..interfaces.flying import flying
from .bird import bird

class wild_hen(bird, flying):
    def __init__(self, name, height, weight, coloreye, flying_height):
        super().__init__(name, height, weight, coloreye, flying_height)

    def voice(self):
       print(f"{super().voice()} ко-ко-ко")

    def fly(self):
        flying.fly(self)
    
    def __str__(self) -> str:
        return super().__str__()